scripts/hello_script.ml

```diff
@@ -0,0 +1,113 @@
+ *
+ * alternatives:
+ *  - ocamlscript?
+ *
+ * references:
+ *  - https://discuss.ocaml.org/t/how-to-run-ml-ocaml-file-without-compiling/4311/21
+ *  - https://discuss.ocaml.org/t/running-ocaml-scripts/10228
+ *  - https://discuss.ocaml.org/t/ocaml-scripting-and-ppx-support/10638
+ *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+
+(* This directive allows then to use #require to load opam/findlib packages. *)
+#use "topfind"
+
+(* You need first to 'opam install ppx_deriving'. *)
+#require "ppx_deriving.show"
+
+(* You need first to 'opam install ppx_deriving_cmdliner'. *)
+#require "ppx_deriving_cmdliner"
+
+#load "unix.cma"
+
+#load "threads/threads.cma"
+
+(* You need first to 'opam install feather'.
+ * For some unknown reasons, requiring feather does not load the unix and
+ * threads dependencies, so I had to do it manually above.
+ *)
+#require "feather"
+
+(* This assumes you did a 'make install' of semgrep-core so that
+ * the pfff (and semgrep-core) libraries are available from topfind.
+ *)
+#require "commons"
+
+open Common
+open Cmdliner
+module F = Feather
+open Feather (* for |. *)
+open Feather.Infix (* for >, <, ||., etc. *)
+
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+
+(* see https://github.com/hammerlab/ppx_deriving_cmdliner *)
+type cli_params = {
+  username : string; [@default "pad"]
+  verbose : bool; [@default false]
+  command : string; [@pos 0] [@docv "CMD"]
+}
+[@@deriving cmdliner, show]
+
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+(* see https://github.com/charlesetc/feather *)
+let simple_shell_programming_demo () =
+  let out =
+    F.process "ps" [ "-aux" ]
+    |. F.map_lines ~f:String.uppercase_ascii
+    |. F.process "grep" [ "OCAML" ]
+    (* alt: |. F.grep "OCAML" *)
+    |> F.collect F.stdout
+  in
+  pr out
+
+(*****************************************************************************)
+(* Entry point *)
+(*****************************************************************************)
+
+let run ps =
+  if ps.verbose then pr2 (spf "debug: params = %s" (show_cli_params ps));
+  pr (spf "Hello %s from %s" ps.command ps.username);
+  simple_shell_programming_demo ();
+  ()
+
+(*****************************************************************************)
+(* Cmdliner boilerplate *)
+(*****************************************************************************)
+let main () =
+  let info = Cmd.info Sys.argv.(0) in
+  let term = Term.(const run $ cli_params_cmdliner_term ()) in
+  let cmd = Cmd.v info term in
+  exit (Cmd.eval cmd)
+
+let () = main ()
```

Comment:
We should probably create a library function that takes care of the cmdliner incantations. Something like
```ocaml
(* Take a command-line parser, execute a program, and exit. *)
val main : 'a Cmdliner.Term.t -> ('a -> int) -> 'b
```

Reply:
Such a `main` utility should also provide stack backtrace recording. Maybe also a few other things I'm not thinking of right now.